pallets/parachain-system/proc-macro/src/lib.rs

```diff
@@ -101,6 +102,28 @@ pub fn register_validate_block(input: proc_macro::TokenStream) -> proc_macro::To
 		Err(e) => return e.into_compile_error().into(),
 	};
 
+	let check_inherents = match check_inherents {
+		Some(_check_inherents) => {
+			#[cfg(not(feature = "parameterized-consensus-hook"))]
+			quote::quote! { #_check_inherents }
+			#[cfg(feature = "parameterized-consensus-hook")]
+			return Error::new(
+				Span::call_site(),
+				"CheckInherents is incompatible with consensus hook feature",
```

Comment:
I like this approach to giving a compiler error when trying to use both `CheckInherents` and `ConsensusHook`.